opcplc/PlcNodeManagerFromFile.cs

```diff
@@ -0,0 +1,544 @@
+                {
+                    CreateBaseVariable(folderFromConfigFile, node);
+                    Logger.Information($"{nameof(PlcNodeManagerFromFile)} created node {node.NodeId} - {node.Name}.");
+                }
+                
+                FolderState root = CreateFolder(null, ProgramName, ProgramName);
+                root.AddReference(ReferenceTypes.Organizes, true, ObjectIds.ObjectsFolder);
+                references.Add(new NodeStateReference(ReferenceTypes.Organizes, false, root.NodeId));
+                root.EventNotifier = EventNotifiers.SubscribeToEvents;
+                AddRootNotifier(root);
+
+                AddPredefinedNode(SystemContext, folderFromConfigFile);
+
+                Logger.Information($"{nameof(PlcNodeManagerFromFile)} bootstrap completed.");
+                Logger.Information($"");
+            }
+
+            base.CreateAddressSpace(externalReferences);
+        }
+
+
+
+        /// <summary>
+        /// Sets properies of the ResetTrend method.
+        /// </summary>
+        private void SetResetTrendMethodProperties(ref MethodState method)
+        {
+            method.OnCallMethod = new GenericMethodCalledEventHandler(OnResetTrendCall);
+        }
+
+        /// <summary>
+        /// Sets properies of the ResetStepUp method.
+        /// </summary>
+        private void SetResetStepUpMethodProperties(ref MethodState method)
+        {
+            method.OnCallMethod = new GenericMethodCalledEventHandler(OnResetStepUpCall);
+        }
+
+        /// <summary>
+        /// Sets properies of the StartStepUp method.
+        /// </summary>
+        private void SetStartStepUpMethodProperties(ref MethodState method)
+        {
+            method.OnCallMethod = new GenericMethodCalledEventHandler(OnStartStepUpCall);
+        }
+
+        /// <summary>
+        /// Sets properies of the StopStepUp method.
+        /// </summary>
+        private void SetStopStepUpMethodProperties(ref MethodState method)
+        {
+            method.OnCallMethod = new GenericMethodCalledEventHandler(OnStopStepUpCall);
+        }
+
+        /// <summary>
+        /// Creates a new variable.
+        /// </summary>
+        private BaseDataVariableState CreateBaseVariable(NodeState parent, string path, string name, BuiltInType dataType, int valueRank, byte accessLevel, string description)
+        {
+            return CreateBaseVariable(parent, path, name, (uint)dataType, valueRank, accessLevel, description);
+        }
+
+        private void CreateBaseVariable(FolderState root, ConfigNode node)
+        {
+            BuiltInType tp;
+            if (Enum.TryParse<BuiltInType>(node.DataType, out tp) == false)
+                throw new ArgumentException($"Value {node.DataType} of node {node.NodeId} cannot be parsed.");
+
+            byte accessLevel = 0;
+
+            // We have to hard code conversion here, because AccessLevel is defined as byte in OPCUA lib.
+            switch (node.AccessLevel)
```

Comment:
How about using something like this:
            try
            {
                accessLevel = (byte)(typeof(AccessLevels).GetField(node.AccessLevel).GetValue(null));
            }
            catch (Exception e)
            {
                throw new ArgumentException($"Unsupported AccessLevel '{node.AccessLevel}'");
            }